src/app/transactions/[transactionId]/page.tsx

```diff
@@ -0,0 +1,70 @@
+import NotFoundPage from "@/app/not-found";
+import { MessageSection } from "@/app/transactions/[transactionId]/MessageSection";
+import { findTransaction } from "@/services/transaction";
+import { getSessionUser } from "@/utils/getSession";
+import Image from "next/image";
+import { FaChevronRight } from "react-icons/fa";
+import defaultIcon from "@/assets/profile-pic-placeholder.png";
+import { TransactionStatus } from "@/app/transactions/[transactionId]/TransactionStatus";
+
+/**
+ * 取引画面
+ * @param param0.transactionId 取引ID
+ */
+export default async function Transaction({
+  params,
+}: {
+  params: { transactionId: string };
+}) {
+  const transactionId = params.transactionId;
+  const [transaction, sessionUser] = await Promise.all([
+    findTransaction(transactionId),
+    getSessionUser()
+  ]);
+
+  if (!transaction) {
```

Comment:
NotFoundPageをJSXで返すのではなく、next/navigationのnotFound()関数を呼ぶのが正しい実装だと思います。
また、このようにしたほうが簡潔で良いと思いました。

```suggestion
const sellerId = transaction.listing.sellerId;
const buyerId = transaction.buyerId;

const isNotSellerOrBuyer = sessionUser.id !== sellerId && sessionUser.id !== buyerId;

  if (!transaction || !sessionUser || isNotSellerOrBuyer) {
    notFound()
  }

```

Reply:
なるほどですね
ではnotFound()だけお願いします